crates/handler/src/validation.rs

```diff
@@ -237,7 +238,8 @@ pub fn validate_initial_tx_gas(
 
     // EIP-7623: Increase calldata cost
     // floor gas should be less than gas limit.
-    if spec.is_enabled_in(SpecId::PRAGUE) && gas.floor_gas > tx.gas_limit() {
+    if !is_eip7623_disabled && spec.is_enabled_in(SpecId::PRAGUE) && gas.floor_gas > tx.gas_limit()
```

Comment:
This might not be strictly necessary since the `floor_gas` will be 0 if EIP-7623 is disabled. I just found it more clear to still add it in the condition.

Let me know if you want me to remove it.